src/sentry/snuba/sessions_v2.py

```diff
@@ -315,9 +317,10 @@ def __init__(self, query, params, allowed_resolution: AllowedResolution):
 ONE_HOUR = timedelta(hours=1).total_seconds()
 ONE_MINUTE = timedelta(minutes=1).total_seconds()
 
-#: Snuba applies a default limit of 1000, let's make it explicit here.
+#: Snuba has a maximum limit of 10000, so the number of groups times the number
+#: of time intervals should never exceed that limit.
 #: https://github.com/getsentry/snuba/blob/69862db3ad224b48810ac1bb3001e4c446bf0aff/snuba/query/snql/parser.py#L908-L909
-SNUBA_LIMIT = 1000
+SNUBA_LIMIT = 10000
```

Comment:
@evanh I agree with your comments that sending 10000 data points to a UI rarely makes sense. However, this endpoint is also public API used by external clients, which presumably expect a complete set of data. So I think returning as many groups as possible is the best we can do for that use case.

Reply:
I don't understand that reasoning. Returning less data points doesn't make the data incomplete, just less granular (e.g. daily instead of hourly).

If this endpoint is going to be a public API then we should be more restrictive by default, instead of less restrictive.

Finally while this will be a public API, it's purpose is to power our UI and we should design it with that in mind. Returning that many data points doesn't improve the UI and might even degrade performance since it has to discard/render a bunch of useless points.